### ISSUE TITLE
Enhancement to Kobalt server: return dependencies like a tree hierarchy

### ISSUE THREAD

**Issue by dmitry-zhuravlev** [cbeust/kobalt#238]
@cbeust  At the moment Kobalt server return flat dependencies. It would be better if Kobalt server can return dependencies like a tree. IDEA plugin could group this dependencies in the tool window. E.g.
![image](https://cloud.githubusercontent.com/assets/6297697/15862089/376f104e-2cd7-11e6-9ff4-aff0ba045a46.png)

It brings to user much better view of dependencies and which dependency from which one comes transitively.

**Comment by dmitry-zhuravlev** [comment]
ping @cbeust Any thoughts about it?

**Comment by cbeust** [comment]
Sorry forgot about this. Yes, I'm open to that, I should be able to take a look in the next few days.

**Comment by cbeust** [comment]
Now in Kobalt 0.880.

You still receive `DependencyData` but that class has an additional `children` constructor parameter:

```
    class DependencyData(val id: String, val scope: String, val path: String,
            val children: List<DependencyData> = emptyList())
```

The WebSocket endpoint is `"/v1/getDependencyGraph"`.

I also updated the structure un the `kobalt-intellij-plugin`, you just need to call the new endpoint and walk the children tree.

Let me know when you have something for me to try.

**Comment by dmitry-zhuravlev** [comment]
Thanks! I'll take a look and implement hierarchy in the plugin.

However I found some maybe bug:
If I specify  
`dependencies {
  provided("...")
  runtime("...")
}`

It still sended in DependencyData.scope like "compile". "runtime" doesn't sended as well.

Also in Kobalt you can specify something like:
` dependenciesTest {
   provided("...")
   runtime("...")
}`

But in IDEA you cannot do it. The test scope dependencies cannot be provided or runtime in the same time. Is it makes sense in Kobalt to have such provided\runtime test dependencies?

**Comment by cbeust** [comment]
Yes, I only return the `compile` dependencies for now. It's trivial to fix:

https://github.com/cbeust/kobalt/blob/master/src/main/kotlin/com/beust/kobalt/app/remote/DependencyData.kt#L68-L71

**Comment by dmitry-zhuravlev** [comment]
I think it is not so trivial because `IClasspathDependency` knows nothing about the scope. And one of the implementations like `AetherDependency` should be extended to incapsulate the scope. 
@cbeust Maybe you see some more simpler solution?

**Comment by cbeust** [comment]
But what the server sends you is not `IClasspathDependency`, it's `DependencyData`, which includes the scope:

```
    class DependencyData(val id: String, val scope: String, val path: String,
            val children: List<DependencyData> = emptyList())
```

**Comment by dmitry-zhuravlev** [comment]
`DependencyData` have only hardcoded `compile` like described here:
https://github.com/cbeust/kobalt/blob/master/src/main/kotlin/com/beust/kobalt/app/remote/DependencyData.kt#L68-L71

So to eliminate this hardcode we need to get the scope from some other place...
